virttest/utils_conn.py

```diff
@@ -1105,18 +1126,27 @@ def server_setup(self, on_local=False):
             except remote.SCPError as detail:
                 raise ConnSCPError('AdminHost', local_path,
                                    server_ip, remote_path, detail)
-        # When qemu supports TLS, it needs not to modify below
-        # configuration files, so simply return.
-        if self.qemu_tls:
-            return
 
         # Ensure to use proper configuration objects
         if on_local:
             operate_libvirtdconf = self.client_libvirtdconf
             operate_syslibvirtd = self.client_syslibvirtd
+            operate_qemuconf = self.server_qemuconf
         else:
             operate_libvirtdconf = self.server_libvirtdconf
             operate_syslibvirtd = self.server_syslibvirtd
+            operate_qemuconf = self.client_qemuconf
+
+        # Change qemu conf file to support tls for chardev
+        if self.qemu_tls:
+            pattern2repl = {r".*chardev_tls\s*=\s*.*":
+                            "chardev_tls = 1"}
+            operate_qemuconf.sub_else_add(pattern2repl)
+            pattern2repl = {r".*chardev_tls_x509_cert_dir\s*="
+                            "\s*\"\/etc\/pki\/libvirt-chardev\s*\".*":
+                            "chardev_tls_x509_cert_dir="
+                            "\"/etc/pki/libvirt-chardev\""}
+            operate_qemuconf.sub_else_add(pattern2repl)
```

Comment:
As you remove line 1108 ~ 1111,  line 1152 and after will be executed. This may break the correct configuration for --tls case (virsh.migrate_options_shared..native_tls).  I remember we should not add --listen in /etc/sysconfig/libvirtd when using --tls for migration.  Please take care and double check to make sure the existing case PASS. Thanks.

Reply:
Line 1152 is not removed and the --listen is not added. This change doesn't include anything about the libvirtd.  It's to config /etc/libvirt/qemu.conf as console cases require.

Reply:
@jinqi7 , In migrate_options_shared..native_tls, qemu_tls indicates the migration will use --tls option which is to use the TLS supported by qemu, not libvirt as before, which do not need to edit the /etc/sysconfig/libvirtd to add '--listen' args in libvirtd configuration, otherwise the migration will fail. You can try to run that case locally with your change to see if side effect there.